salvo/src/lib/cmd_exec.py

```diff
@@ -35,18 +36,32 @@ def run_command(cmd: str, parameters: CommandParameters) -> str:
       command
   """
   output = ''
+  params = parameters._asdict()
+  tmpfile = tempfile.TemporaryFile(
+      mode='w+', dir=params['cwd'], prefix='cmd_output'
+  )
+
   try:
     log.debug(f"Executing command: [{cmd}] with args [{parameters._asdict()}]")
     cmd_array = shlex.split(cmd)
-    output = subprocess.check_output(
-        cmd_array, stderr=subprocess.STDOUT, **parameters._asdict())
 
-    if isinstance(output, bytes):
-      output = output.decode('utf-8').strip()
+    subprocess.check_call(cmd_array, stdout=tmpfile, stderr=tmpfile,
+                          **parameters._asdict())
```

Comment:
(nit) Can we use `params` here?

**params that is.

Reply:
With this change I get:
```
TypeError: check_call() argument after ** must be a mapping, not CommandParameters
```

Reply:
Thank you trying it out.